frappe/integrations/doctype/google_calendar/google_calendar.py

```diff
@@ -186,17 +194,28 @@ def get_authentication_url(client_id=None, redirect_uri=None):
 		"url": (
 			"https://accounts.google.com/o/oauth2/v2/auth?"
 			f"access_type=offline&response_type=code&prompt=consent&client_id={client_id}"
-			f"&include_granted_scopes=true&scope={SCOPES}&redirect_uri={redirect_uri}"
+			f"&include_granted_scopes=true&scope={SCOPES}&redirect_uri={quote(redirect_uri, safe='')}"
 		)
 	}
 
 
-@frappe.whitelist()
+@frappe.whitelist(allow_guest=True)
 def google_callback(code=None):
 	"""
-	Authorization code is sent to callback as per the API configuration
+	Authorization code is sent to callback as per the API configuration.
+	OAuth callbacks need allow_guest=True since user is not authenticated during redirect.
```

Comment:
This shouldn't be needed, the user would have to be logged in to trigger this in the first place right?